src/test/java/fr/lunatech/timekeeper/timeutils/TimeKeeperDateUtilsTest.java

```diff
@@ -329,4 +336,302 @@ void shouldReturnMondayAsFirstDayOfWeek() {
+    })
+    void shouldComputeDuration(int to, int from, Long expected, ChronoUnit unit) {
+        LocalDateTime start = LocalDateTime.of(2020, 10, 6, to, 0);
+        LocalDateTime end = LocalDateTime.of(2020, 10, 6, from, 0);
+        assertEquals(expected, TimeKeeperDateUtils.getDuration(start, end, unit));
+    }
+
+    @Test
+    void shouldThrowAnErrorIfStartIsAfterEnd() {
+        LocalDateTime start = LocalDateTime.of(2020, 10, 6, 10, 0);
+        LocalDateTime end = LocalDateTime.of(2020, 10, 6, 9, 0);
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            TimeKeeperDateUtils.getDuration(start, end, ChronoUnit.MINUTES);
+        });
+    }
+
+    @Test
+    void shouldReturnTrueIfItsSameWeekAndSameYear() {
+        LocalDate date1 = LocalDate.of(2020, 10, 6);
+        LocalDate date2 = LocalDate.of(2020, 10, 8);
+        assertTrue(TimeKeeperDateUtils.isSameWeekAndYear(date1, date2));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            //end year, end day
+            "2021, 20", //shouldReturnFalseIfItsDifferentYearAndWeek
+            "2021, 8", //shouldReturnFalseIfItsSameWeekButDifferentYear
+            "2020, 20" //shouldReturnFalseIfItsSameYearButDifferentWeek
+    })
+    void shouldReturnFalseIfNotSameWeekOrYear(int endYear, int endDay) {
+        LocalDate date1 = LocalDate.of(2020, 10, 6);
+        LocalDate date2 = LocalDate.of(endYear, 10, endDay);
+        assertFalse(TimeKeeperDateUtils.isSameWeekAndYear(date1, date2));
+    }
+
+    @Test
+    void shouldReturn1ForJanuary() {
+        LocalDate date = LocalDate.of(2020, 1, 20);
+        assertEquals(1, TimeKeeperDateUtils.getMonthNumberFromDate(date));
+    }
+
+    @Test
+    void shouldReturnALocalDateTimeFromAString() {
+        String date = "2020-10-06T09:00:00";
+        assertEquals(
+                LocalDateTime.of(2020, 10, 6, 9, 0, 0),
+                TimeKeeperDateUtils.formatToLocalDateTime(date));
+    }
+
+    @Test
+    void shouldReturnALocalDateFromAString() {
+        String date = "2020-06-11";
+        assertEquals(
+                LocalDate.of(2020, 6, 11),
+                TimeKeeperDateUtils.formatToLocalDate(date));
+    }
+
+    @Test
+    void shouldCompute0Hours() {
+        assertEquals(0L,
+                TimeKeeperDateUtils.computeTotalNumberOfHours(LocalDate.of(2020, 6, 17).atTime(9, 0).truncatedTo(ChronoUnit.HOURS),
+                        LocalDate.of(2020, 6, 17).atTime(9, 0).truncatedTo(ChronoUnit.HOURS)
+                )
+        );
+    }
+
+    @Test
+    void shouldCompute3BusinessHours() {
+        assertEquals(3L,
+                TimeKeeperDateUtils.computeTotalNumberOfHours(LocalDate.of(2020, 6, 17).atTime(9, 0).truncatedTo(ChronoUnit.HOURS),
+                        LocalDate.of(2020, 6, 17).atTime(12, 0).truncatedTo(ChronoUnit.HOURS)
+                )
+        );
+    }
+
+    @Test
+    void shouldCompute8BusinessHours() {
+        assertEquals(8L,
```

Comment:
For all tests part, you should create a helper test class, to generate test data, look at  DataTestProvider